### PR TITLE
[feat-5266]: kto over iets anders

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-5266](https://gemeente-amsterdam.atlassian.net/browse/SIG-5266)

To enable open question for the KTO process I had to change the checkbox and radiobutton lists. 

### Changes
- render textarea when a checkbox or radiobutton option has the open_answer property
- Add form methods that updates the form and validates the input
- Remove the hardcoded "over iets anders"-logic
- Implement utils that map over the answers and pair the correct checked box with the open answer
- Add tests

### CODE REVIEW

File: src/components/RadioButtonList/RadioButtonList.tsx [hunk 1]
```diff
@@ -90,27 +108,67 @@ const RadioButtonList = <T extends RadioButtonOption>({
...
+                formValidation.errors[`open_answer-${option.value}`]?.type ===
+                errorType
+              }
+              getError={(errorType) =>
+                formValidation.errors[`open_answer-${option.value}`]?.type ===
+                errorType
+              }
+            >
+              <TextArea
+                {...formValidation.register(`open_answer-${option.value}`)}
```
Comment (OscarBakker) [inline src/components/RadioButtonList/RadioButtonList.tsx:157]: Is the set value necessary and the trigger, I guess register already adds there through descructuring.

Comment (OscarBakker) [reply src/components/RadioButtonList/RadioButtonList.tsx:157]: I see that onchange does a thing under the hood, its checks the max content and shows chars left do never mind

File: src/components/RadioButtonList/RadioButtonList.tsx [hunk 2]
```diff
@@ -139,7 +197,9 @@ const RadioButtonList = <T extends RadioButtonOption>({
         id={id}
         {...rest}
       >
-        {radioOptions.map((option, index) => renderRadioButton(option, index))}
+        {radioOptions.map((option, index) =>
+          renderRadioButton(option, index, formValidation)
```
Comment (OscarBakker) [inline src/components/RadioButtonList/RadioButtonList.tsx:201]: I see this was already a function, but would it not be better just add jsx here? or make a special Button component within RAdioButtonList/

File: src/signals/incident/containers/KtoContainer/utils.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    if (index !== -1) {
+      resultArray.splice(index + 1, 0, value)
+    }
+  })
+
+  return resultArray
+}
+
+export const getMergedOpenAnswers = (object: FormData): FormData => {
+  const openAnswerPairs: Record<string, any> = {}
```
[the remark this answers is not in the document]
Comment (vdegraaf) [reply src/signals/incident/containers/KtoContainer/utils.ts:22]: This gives some typing errors, so left it as is.

File: src/components/RadioButtonList/RadioButtonList.tsx (hunk 2 above)
Comment (vdegraaf) [reply src/components/RadioButtonList/RadioButtonList.tsx:201]: You're right!

**Review by OscarBakker — APPROVED** [review]

**Pull request merged**